Updates button usage docs for clarity (#175)
primer/design: content/ui-patterns/button-usage.mdx

| Small     | Decrease button size when space is limited and the action is of lesser significance.                                                                             |
| Large     | Increase button size to bring prominence to buttons that have major impact. Use large buttons sparingly, as layouts should generally have only one large button. |

### Icons

Use [Octicons](https://primer.style/octicons/) in conjunction with buttons to clarify an action if text alone is not enough to explain its functionality.

In some cases, Octicons are used to create an identifiable visual indicator for a common action or paradigm. For example, the repo icon is used next to repo links, the new repo button, and repeated in the new repo flow. This repetition and continuity helps people learn and identify our visual language.

#### Icon with label

Most of the time, a button will have an icon and a label for clarity.

![button with label and icon](https://user-images.githubusercontent.com/2313998/129626583-22dce7ca-611f-42c7-9a51-2ba1551bb5f5.png)

#### Icon only

In some cases, an action can be understood without using just an icon. Icon-only buttons can help save space in a cramped area.

> Buttons that only use an icon should still have a label to make them accesssible to assistive technology (for example: read aloud by a screenreader)

Reviewer: Maybe we should clarify that the label will be invisible to sighted visitors

![button with icon but no label](https://user-images.githubusercontent.com/2313998/129626586-378e2b7f-7300-4804-93fb-c88f7f22407e.png)

### Counters

> Counts can be added as numerical aid in appropriate scenarios by providing valuable context. For example, showing social proof by displaying the number of people watching a repo.

Reviewer: This sentence is a bit wordy
```suggestion
Counters can be added in appropriate scenarios to provide context. For example, showing social proof by displaying the number of people watching a repo.
```

#### Counter inside button

A count can be displayed inside of a button using a [CounterLabel](https://primer.style/react/CounterLabel).

![button with count inside](https://user-images.githubusercontent.com/2313998/129626580-465111c8-9237-4e33-af42-b2e830ea9cc3.png)

#### Counter attached to a button

A count can be displayed as a separate element that is attached to a button. Only display a count as separate element that is attached to a button when the button uses the "small" size variant and the button type is "Default" or "Outline".

![button with count attached](https://user-images.githubusercontent.com/2313998/129626579-a2f17327-1b6e-43b4-8971-5b9a24f96dda.png)

> ### Adaptable full-width

Reviewer: ```suggestion
### Full-width
```
